### PR TITLE
Add Vuex and store user information in 'store'

### PR DESCRIPTION
**Pull request by VeerpalBrar**
- install vuex to the repo
- update the Vuex store to keep track of if the user is logged in, the user's utorid and the user's id so that this is available to all users
- update the user information on each route change (so it's updated when the user is redirected from UofT weblogin to ConnectEd)

### CODE REVIEW

File: client/src/components/Projects.vue [hunk 1]
```diff
@@ -52,7 +51,7 @@
         },
         methods:{
             isLoggedIn(){
-                return AuthenticationService.userIsLoggedIn(this.utorid);
+                return this.$store.state.user.utorid == this.utorid
```
Comment (VeerpalBrar) [inline client/src/components/Projects.vue]: get's the current logged in users id from the vuex store and compare to the utorid of the current profile. If they match we want to display the "Add Project" button

File: client/src/router/index.js [hunk 2]
```diff
@@ -93,4 +94,9 @@ const router = new VueRouter({
     routes
 });
 
+router.beforeEach((to, from, next) => {
```
Comment (VeerpalBrar) [inline client/src/router/index.js]: Before a route change, update user info. This way, when user is redirected from web login page to our site, we update the user info.

File: server/src/controllers/RedirectController.js [hunk 3]
```diff
@@ -6,14 +6,15 @@ module.exports = {
...
-        res.sendFile(path.join(__dirname, '../dist/index.html'))
+        res.redirect('/connect/404')
       }
       user = await User.findOne({
         where: {
           utorid: req.headers.utorid
         }
       })
       if (user){
+        res.cookie('id', user.id)
```
Comment (VeerpalBrar) [inline server/src/controllers/RedirectController.js]: If the user exists, let store the user id in a cookie as well, for convenience.

File: client/src/store.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      state.user = user
+    },
+    setLoggedIn(state, val){
+      state.isLoggedIn = val
+    }
+  },
+  actions: {
+    setUser ({commit}) {
+      var user = {}
+      if  (AuthenticationService.isLoggedIn()){
```
Comment (VeerpalBrar) [inline client/src/store.js]: if the `utorid` and `id` cookies are present, let's store this information within the Vuex store for easy access. Also, set the `isLoggedIn` value to `true`

File: client/src/router/index.js (hunk 2 above)
Comment (ishaaa-ai) [reply client/src/router/index.js]: this is a question. is the user information updated on line 98 when the route is changed?

**Pull request merged**